programs/cleos/main.cpp

```diff
@@ -436,12 +432,61 @@ fc::variant push_transaction( signed_transaction& trx, const std::vector<public_
       if (tx_use_old_rpc) {
          return call(push_txn_func, packed_transaction_v0(trx, compression));
       } else {
-         try {
-            return call(send_txn_func, packed_transaction_v0(trx, compression));
-         }
-         catch (chain::missing_chain_api_plugin_exception &) {
-            std::cerr << "New RPC send_transaction may not be supported. Add flag --use-old-rpc to use old RPC push_transaction instead." << std::endl;
-            throw;
+         if( !amqp_address.empty() ) {
+            transaction_trace_msg trace_msg;
+            eosio::transaction_msg msg{packed_transaction( std::move( trx ), true, compression )};
+            auto buf = fc::raw::pack( msg );
+            const auto& tid = msg.get<packed_transaction>().id();
+            const string id = tid.str();
+            {
+               eosio::chain::named_thread_pool trx_thread_pool( "mqtrx", 1 );
+               eosio::chain::named_thread_pool trace_thread_pool( "mqtrac", 1 );
+               eosio::amqp qp_trx( trx_thread_pool.get_executor(), amqp_address, "trx", [](const std::string& err){
+                  std::cerr << "AMQP trx error: " << err << std::endl;
+                  exit(1);
+               } );
+               eosio::amqp qp_trace( trace_thread_pool.get_executor(), amqp_address, "trace", [](const std::string& err){
+                  std::cerr << "AMQP trace error: " << err << std::endl;
+                  exit(1);
+               } );
+
+               qp_trx.publish( "", id, buf.data(), buf.size() );
+
+               std::promise<void> received_trace;
+               std::atomic_bool received = false;
+               auto& consumer = qp_trace.consume();
+               consumer.onError( []( const char* message ) {
+                  std::cerr << "consume failed: " << message << std::endl;
+               } );
+               consumer.onReceived( [&]( const AMQP::Message& message, uint64_t delivery_tag, bool redelivered ) {
+                  if( message.hasCorrelationID() && message.correlationID() == id && !received) {
+                     fc::datastream<const char*> ds( message.body(), message.bodySize() );
+                     fc::raw::unpack( ds, trace_msg );
+                     // don't ack as other consumers may be interested in trace: qp_trace.ack( delivery_tag );
```

Comment:
Since the goal is for amqp trx to `nodeos` to be async with possible multiple readers, modified `cleos` to only submit trx and not try to consume traces.